spec/helpers/perf.lua

```diff
@@ -125,13 +128,48 @@ function _M.start_upstreams(conf, port_count)
   return invoke_driver("start_upstreams", conf, port_count)
 end
 
+local function dp_conf_from_cp_conf(kong_conf)
+  local dp_conf = {}
+  for k, v in pairs(kong_conf) do
+    dp_conf[k] = v
+  end
+  dp_conf['role'] = 'data_plane'
+  dp_conf['database'] = 'off'
+  dp_conf['cluster_control_plane'] = 'kong-cp:8005'
+  dp_conf['cluster_telemetry_endpoint'] = 'kong-cp:8006'
+
+  return dp_conf
+end
+
+--- Start Kong in hybrid mode with given version and conf
+-- @function start_hybrid_kong
+-- @param version string Kong version
+-- @param kong_confs table Kong configuration as a lua table
+-- @return nothing. Throws an error if any.
+function _M.start_hybrid_kong(version, kong_confs)
+  if DRIVER_NAME ~= 'docker' then
+    error("Hybrid support only availabe in Docker driver")
+  end
+  local kong_confs = kong_confs or {}
+
+  kong_confs['cluster_cert'] = '/kong_clustering.crt'
+  kong_confs['cluster_cert_key'] = '/kong_clustering.key'
+  kong_confs['role'] = 'control_plane'
+
+  local control_plane = _M.start_kong(version, kong_confs)
+  local driver_confs = { dns = { ['kong-cp'] = control_plane } }
+  _M.start_kong(version, dp_conf_from_cp_conf(kong_confs), driver_confs)
+end
+
 --- Start Kong with given version and conf
 -- @function start_kong
 -- @param version string Kong version
 -- @param kong_confs table Kong configuration as a lua table
+-- @param driver_confs table driver configuration as a lua table
 -- @return nothing. Throws an error if any.
-function _M.start_kong(version, kong_confs)
-  return invoke_driver("start_kong", version, kong_confs)
+function _M.start_kong(version, kong_confs, driver_confs)
+  DATA_PLANE = invoke_driver("start_kong", version, kong_confs or {}, driver_confs or {})
+  return DATA_PLANE
```

Comment:
this breaks the non hybrid use case as the `DATA_PLANE` variable would never be set. i agree that's ugly setting DATA_PLANE directly here, but it's super convenient as the last kong instance ever started is promoted to data plane.

Reply:
for non-hybrid deployment and there's only one kong container, the driver don't need to tell which one to
send_load to. otherwise i would rather return the data_plane ID to upper level and let user select, since
depending on the order to distinguish CP and DP is bit too implict